routes/relationships.js

```diff
@@ -0,0 +1,252 @@
+        let mentorList = await users.getMentorList(userId);
+        returnList.concat(await relationships.filterRelationshipsByStatus(mentorList, req.params.status));
+        
+        let relationshipObjects = []
+        for (let relationshipId of returnList){
+            relationshipObjects.push(await relationships.getRelationshipById(relationshipId));
+        };
+        //return relationship objects
+        //res.render('frames/relationships', {layout: 'profile', relationships: relationshipObjects});
+
+        res.status(200).json({success: true, relationships: returnObjects});
+    }
+    catch(e) {
+        //res.status(500).render('frames/error', {layout: 'profile', error: "Internal server error"});
+        res.status(500).json({error:e});
+    }
+}
+
+/**
+ * Get all relationships where current user is the mentee
+ * @param {Object} req - request object
+ * @param {Object} res - response object
+ */
+async function getMentors(req, res){
+    let returnList;
+    let userId;
+
+    try {
+        userId = validate.convertID(req.params.userId);
+        let mentorRelationships = await users.getMentorList(userId);
+        
+        let relationshipObjects = []
+        for (let relationshipId of mentorRelationships){
+            relationshipObjects.push(await relationships.getRelationshipById(relationshipId));
+        };
+        //return relationship objects
+        //res.render('frames/relationships', {layout: 'profile', relationships: relationshipObjects});
+        res.status(200).json({success: true, relationships: relationshipObjects});
+    }
+    catch(e) {
+        //res.status(500).render('frames/error', {layout: 'profile', error: "Internal server error"});
+        res.status(500).json({error:e});
+    }
+};
+
+/**
+ * Get all relationship objects where current user is mentor
+ * @param {Object} req 
+ * @param {Object} res 
+ */
+async function getMentees(req, res){
+    let returnList;
+    let userId;
+
+    try {
+        userId = validate.convertID(req.params.userId);
+        let menteeRelationships = await users.getMenteeList(userId);
+        let relationshipObjects = []
+        for (let relationshipId of menteeRelationships){
+            relationshipObjects.push(await relationships.getRelationshipById(relationshipId));
+        };
+        //return relationship objects
+        //res.render('frames/relationships', {layout: 'profile', relationships: relationshipObjects});
+
+        res.status(200).json({success: true, relationships: relationshipObjects});
+        
+    }
+    catch(e) {
+        //res.status(500).render('frames/error', {layout: 'profile', error: "Internal server error"});
+        res.status(500).json({error:e});
+    }
+};
+
+/**
+ * TODO: Define a route for this function
+ * Update relationship object 
+ * Needs the relationshipID, and new status
+ * @param {Object} req - request object
+ * @param {Object} res - response object
+ */
+ async function postRelationshipUpdate(req, res){

```

Comment:
URL should be something like 
- /relationships/${relationID}/status if you want to take the status from body
- /relationships/${relationID}/status/${status} 

Any mandatory parameter that is representing the resource we are using can be passed as either PathParameter or RequestBody.